fapolicy_analyzer/ui/fapd_manager.py

```diff
@@ -0,0 +1,215 @@
+# Copyright Concurrent Technologies Corporation 2021
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import logging
+import os
+import subprocess
+
+
+from datetime import datetime as DT
+from enum import Enum
+from fapolicy_analyzer import Handle
+from threading import Lock
+from time import time, sleep
+
+
+class ServiceStatus(Enum):
+    FALSE = False
+    TRUE = True
+    UNKNOWN = None
+
+
+class FapdMode(Enum):
+    DISABLED = 0
+    ONLINE = 1
+    PROFILING = 2
+
+
+class FapdManager():
+    def __init__(self, bSuFapdControl=True):
+        logging.debug("FapdManager::__init__(self)")
+        self._fapd_status = ServiceStatus.UNKNOWN
+        self._fapd_monitoring = False
+        self._fapd_ref = Handle("fapolicyd")
+        self._fapd_profiler_pid = None
+        self._fapd_control_enabled = bSuFapdControl
```

Comment:
It will. Previously calls to the fapd backend functions were restricted at the MainWindow level by graying out the menu items for non-root users.

Wired it up so that self.mode can not be modified unless an env var, SU_OVERRIDE is set, so that we can modify the mode during development. Eventually SU_OVERRIDE should be removed when the project has a production release.